sklearn/impute/_iterative.py

```diff
@@ -559,7 +559,8 @@ def fit_transform(self, X, y=None):
             self._estimator = clone(self.estimator)
 
         if hasattr(self._estimator, 'random_state'):
-            self._estimator.random_state = self.random_state_
+            if self._estimator.random_state is None:
+                self._estimator.random_state = self.random_state_.randint(np.iinfo(np.int32).max)
```

Comment:
Linting is failing because the line is too long

Reply:
also this should probably be using `set_params`. we might special-case the way we handle random_state in the future as discussed in https://github.com/scikit-learn/scikit-learn/issues/14042